File: Backend/outfit/models/Outfit.py
from django.contrib.postgres.fields import ArrayField
from django.db import models
from item.models.Item import Item


class Outfit(models.Model):
    id = models.AutoField(primary_key=True)
    items = ArrayField(Item)
    rating = models.IntegerField()
    date = models.DateField()  # format-> YYYY-MM-DD

File: frontend/lib/Models/outfit.dart
import 'package:frontend/Models/item.dart';

class Outfit {
  List<Item> items;
  int rating;
  String date; // Represented as a string for simplicity

  Outfit({
    required this.items,
    required this.rating,
    required this.date,
  });

  factory Outfit.fromJson(Map<String, dynamic> json) {
    return Outfit(
      items: List<Item>.from(json['items']),
      rating: json['rating'],
      date: json['date'],
    );
  }
}
